src/app/pages/login/login.component.ts

```diff
@@ -1,59 +1,70 @@
 import { Component } from '@angular/core';
-import { FormBuilder, FormGroup, FormsModule, ReactiveFormsModule, Validators } from '@angular/forms';
-import { Router, RouterLink, RouterLinkActive } from '@angular/router';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { Router, RouterLink, RouterLinkActive, NavigationStart } from '@angular/router';
 import { AuthService } from '../../auth/auth.service';
-import { HttpClient, HttpHeaders, HttpErrorResponse } from '@angular/common/http';
-import { Observable, throwError, catchError, of } from 'rxjs';
-
-interface LoginResponse{
-  token?: string;
-}
+import { SharedModule } from '../../shared/shared.module';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { Subscription } from 'rxjs';
+import { defaultSnackbarConfig } from '../../shared/configs/defaultSnackbarConfig';
 
 @Component({
   selector: 'app-login',
   standalone: true,
-  imports: [RouterLink, RouterLinkActive, FormsModule, ReactiveFormsModule],
+  imports: [RouterLink, RouterLinkActive, SharedModule],
   templateUrl: './login.component.html',
   styleUrl: './login.component.css'
 })
 export class LoginComponent {
   loginForm: FormGroup;
-  private apiUrl = 'https://rb-messenger.azurewebsites.net';
+  isInProgress: boolean = false;
 
-  constructor(private fb: FormBuilder, private authService: AuthService, private http: HttpClient, private router: Router) {
-    this.loginForm = this.fb.group({
+  private _routerSubscription!: Subscription;
+
+  constructor(private _fb: FormBuilder, private _authService: AuthService, private _router: Router,
+    private _snackBar: MatSnackBar
+  ) {
+    this.loginForm = this._fb.group({
       username: ['', Validators.required],
       password: ['', Validators.required]
     });
   }
 
+
+  ngOnInit() {
+    this._routerSubscription = this._router.events.subscribe(event => {
+      if (event instanceof NavigationStart) {
+        this._snackBar.dismiss();
+      }
+    });
+  }
+
   onSubmit() {
     if (this.loginForm.valid) {
       console.log("Submitted login request");
+      this.isInProgress = true;
 
       const username = this.loginForm.value.username;
       const password = this.loginForm.value.password;
 
-      this.login(username, password);
+      this._authService.login(username, password).subscribe(
+        response => {
+          if(response.body && response.body.token){
+            this._authService.setToken(response.body.token);
```

Comment:
`this._authService.setToken(response.body.token);`
This action should be called in AuthService in `.pipe(tap(() => {}))` operator